launch_testing/launch_testing/__init__.py

```diff
@@ -61,13 +63,10 @@ def on_stdout_lines(self, lines):
         if self.regex_match and not self.matched:
             self.matched = re.search(self.expected_output, self.stdout_data.getvalue())
 
-        if self.matched:
-            # We have enough output to compare; shut down my child
+        if self.matched and self.test_process:
+            # We matched and we're in charge; shut it all down
             for td in self.launch_descriptor.task_descriptors:
-                if td.name == self.name:
-                    td.terminate()
-                    self.killed = True
-                    return
+                td.terminate()
```

Comment:
Oh, right, they exit with -15 when we call `terminate()`, which means that we can't use the `primary_exit_handler`, which demands a return code of 0. Let's leave it like this.